cmake/app/boilerplate.cmake

```diff
@@ -486,6 +486,9 @@ if(DTC_OVERLAY_FILE)
   # variable DTC_OVERLAY_FILE
 elseif(DEFINED ENV{DTC_OVERLAY_FILE})
   set(DTC_OVERLAY_FILE $ENV{DTC_OVERLAY_FILE})
+elseif((DEFINED BOARD_REVISION) AND
```

Comment:
I think we should append together, so that you may still have settings common for all revisions of the board in one file and then revision specific overlays in another, as it is already done for shields:
https://github.com/zephyrproject-rtos/zephyr/blob/0426a5a682e5c1b9620042f0410d6c7f1b10cc27/cmake/extensions.cmake#L1938

https://github.com/zephyrproject-rtos/zephyr/blob/0426a5a682e5c1b9620042f0410d6c7f1b10cc27/cmake/app/boilerplate.cmake#L359-L360
if possible, it would also be great if this could be using `zephyr_file()` but that is not required for approving this PR.